src/services/class.service.js

```diff
@@ -48,6 +48,20 @@ const deleteClassById = async (classId, teacher) => {
   return _class;
 };
 
+const getStudents = async (classId) => {
+  return Class.find({ _id: classId }).students;
+};
+
+const addStudent = async (classId, userId) => {
+  Class.update({ _id: classId }, { $push: { students: userId } });
```

Comment:
you are pushing the id of the teacher into the class.
Here we need to create a new Student model and add the student to the class

Reply:
you also forgot the `await` before update

Reply:
what if the student has already been created?

Reply:
We have to decide to either accept only a student id here and add it, but i would prefer if we can accept an object here with all the properties of a student, then we check if the student already exists (using the email) and if he does we add the existing students id, otherwise we create a student and add the id of the newly created student

Reply:
then i leave the problem here.